source/runtime.cpp

```diff
@@ -103,6 +103,10 @@ reshade::runtime::runtime() :
 }
 reshade::runtime::~runtime()
 {
+	// ensure to reset the runtime if this is called direcly by the app (using alt+F4 for instance)
+	if(!_worker_threads.empty() || !_techniques.empty())
+		on_reset();
```

Comment:
I'd rather not have this here. It is not save to call `on_reset` here, because the API may have already been destroyed (e.g. OpenGL context was destroyed, so this could crash, because OpenGL function pointers would be invalid).